app/services/support_interface/ministerial_report_export.rb

```diff
@@ -0,0 +1,60 @@
+module SupportInterface
+  class MinisterialReportExport
+    def call
+      results = subject_status_count
+      subject_mapping(results)
+
+      subject_count = Subject::SUBJECTS.count
+
+      rows = Array.new(subject_count, nil)
+
+      subject_count.times do |n|
+        rows[n] = {
+          subject: Subject::SUBJECTS[n],
+          applications: 0,
+          offer_received: 0,
+          accepted: 0,
+          application_declined: 0,
+          application_rejected: 0,
+        }
+      end
+
+      stem_index = Subject::SUBJECTS.index(:stem)
+      ebacc_index = Subject::SUBJECTS.index(:ebacc)
+      secondary_index = Subject::SUBJECTS.index(:secondary)
```

Comment:
If you made `rows` a hash keyed by `subject` you could maybe avoid all the reverse lookups on `SUBJECTS`, e.g.
```
{
  art_and_design: { applications: 1, offer_received: 2, accepted: 3, application_declined: 4, application_rejected: 5 },
  biology: { applications: 10, offer_received: 9, accepted: 8, application_declined: 7, application_rejected: 6 },
  ...
}
```